src/platform.ts

```diff
@@ -136,39 +135,56 @@ export default class PanasonicPlatform implements DynamicPlatformPlugin {
         this.log.error('Login failed. Skipping device discovery.');
         this.noOfFailedLoginAttempts++;
 
-        if (this.noOfFailedLoginAttempts <= MAX_NO_OF_LOGIN_RETRIES) {
-          /**
-           * | Login retry | Delay (in mins) | Total time lapsed
-           * | 1  | 6  | 6
-           * | 2  | 12 | 18
-           * | 3  | 18 | 36
-           * | 4  | 24 | 60
-           * | 5  | 30 | 90
-           * | 6  | 36 | 126
-           * | 7  | 42 | 168
-           * | 8  | 48 | 216
-           * | 9  | 54 | 270
-           * | 10 | 60 | 330 (= 5h 30mins)
-           */
-          const nextRetryDelay = LOGIN_RETRY_BASE_DELAY * this.noOfFailedLoginAttempts;
+        /**
+        * | Login retry | Delay (in mins) | Total time lapsed
+        * | 1  | 6  | 6
+        * | 2  | 12 | 18
+        * | 3  | 18 | 36
+        * | 4  | 24 | 60
+        * | 5  | 30 | 90
+        * | 6  | 36 | 126
+        * | 7  | 42 | 168
+        * | 8  | 48 | 216
+        * | 9  | 54 | 270
+        * | 10 | 60 | 330 (= 5h 30mins)
+        */
+
+        const maxAttempts = this.platformConfig.maxAttempts || 0;
+
+        if (maxAttempts === 0
+          || this.noOfFailedLoginAttempts <= maxAttempts) {
+
+          const nextRetryDelay = Math.min(LOGIN_RETRY_BASE_DELAY
+            * this.noOfFailedLoginAttempts, 60000);
 
           this.log.error(
-            'The Comfort Cloud server might be experiencing issues at the moment. '
-            + `The plugin will try to log in again in ${nextRetryDelay / 60} minutes. `
-            + 'If the issue persists, make sure you configured the correct email and password '
-            + 'and run the latest version of the plugin. '
-            + 'Restart Homebridge when you change your config.',
+            'The Comfort Cloud server might be experiencing issues at the moment.'
+            + `The plugin will try to log in again in ${nextRetryDelay / 60} minutes.`,
           );
 
+          this.log.error('If the issue persists, make sure: '
+            + 'configured is the correct email and password in plugin settings,'
+            + 'field "Emulated Comfort Cloud app version (override)" in settings '
+            + 'is empty or have the latest version of Panasonic Comfort Cloud '
+            + 'from the App Store (like 1.19.0), '
+            + 'the latest version of this plugin is installed, '
+            + 'all terms and conditions after logging into '
+            + 'the Panasonic Comfort Cloud app are accepted. '
+            + 'Restart Homebridge if you change plugin settings.',
+          );
+
+          if ((this.noOfFailedLoginAttempts === 1) || (this.noOfFailedLoginAttempts %5 === 0)) {
+            this.getAppVersion.bind(this);
```

Comment:
@mkz212, why are you binding this conditionally?

Reply:
- The if condition: on first and on every attempt divided by 5 so : 1,5,10,15 etc. it will run check for new version. 
- From what I remember, I copied what was already there. If something is wrong, please correct.

Reply:
I know the what, but was asking for the why. But I think I figured out what you were trying to do and adjusted it in the latest commit. When trying to log in, the App Store version will be fetched before the first login, and subsequently after every fifth failed attempt.

Reply:
Oh, that's it! It's even a much better solution! Thanks! 👍